core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -57,43 +60,40 @@ public KitRequestService(KitRequestDao dao,
+        this.daoUtils = daoUtils;
     }
 
     /**
      * Send a request for a sample kit to Pepper.
-     *
-     * 1. Create and save the sample kit request in the Juniper database
-     * 2. Send the sample kit request to Pepper
-     * 3. Record the status response from Pepper in the Juniper database
-     *
-     * These steps are all performed in a single transaction. However, each could fail independently. If there's an
-     * error from Pepper, then the sample kit request will not be saved in Juniper, and it's safe to assume that there
-     * is no new request in Pepper. If there's an error saving the status response from Pepper, then it's possible for
-     * there to be a request in Pepper that does not exist in Juniper. We could potentially split that step into its own
-     * transaction to help avoid that potential inconsistency.
+     * Throws PepperApiException if the Pepper API request failed
      */
-    @Transactional
-    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName) {
+    public KitRequest requestKit(AdminUser adminUser, String studyShortcode, Enrollee enrollee, String kitTypeName)
+    throws PepperApiException {
         // create and save kit request
-        Profile profile = profileService.loadWithMailingAddress(enrollee.getProfileId())
-                .orElseThrow(() -> new RuntimeException("Missing profile for enrollee: " + enrollee.getShortcode()));
+        if (enrollee.getProfileId() == null) {
+            throw new IllegalArgumentException("No profile for enrollee: " + enrollee.getShortcode());
+        }
+        Profile profile = profileService.loadWithMailingAddress(enrollee.getProfileId()).get();
         PepperKitAddress pepperKitAddress = makePepperKitAddress(profile);
-        var kitRequest = createKitRequest(adminUser, enrollee, pepperKitAddress, kitTypeName);
+        KitRequest kitRequest = assemble(adminUser, enrollee, pepperKitAddress, kitTypeName);
 
         // send kit request to DSM
-        var response = pepperDSMClient.sendKitRequest(studyShortcode, enrollee, kitRequest, pepperKitAddress);
-
-        // save DSM response/status with Juniper KitRequest
         try {
-            var untypedStatuses = PepperKitStatusResponse.extractUntypedKitStatuses(response, objectMapper);
-            // Pepper response from requesting it kit must have one and only one kit status
-            var pepperStatusJson = objectMapper.writeValueAsString(untypedStatuses.get(0));
+            PepperKitStatus dsmKitStatus = pepperDSMClient.sendKitRequest(studyShortcode, enrollee, kitRequest, pepperKitAddress);
+            // write out the PepperKitStatus as a string for storage
+            var pepperStatusJson = objectMapper.writeValueAsString(dsmKitStatus);
             kitRequest.setDsmStatus(pepperStatusJson);
-            saveKitStatus(kitRequest, pepperStatusJson, kitRequest.getCreatedAt());
+        } catch (PepperParseException e) {
```

Comment:
Hmm... I'm not sure what we'd do with it.  Especially thinking that this is a service that will later on be invoked by automatic processes, I think we want to leave it up to the caller how to handle the various flavors of PepperApiException.

Reply:
But what is a different flavor? Just in terms of encapsulation, why would code external to this service need to understand how to interpret responses from Pepper, particularly if they arrive in the same packaging, PepperApiException.

Reply:
Ah, maybe the code/comments aren't clear about the control flow -- we need to catch the PepperParseException because that means that the request to Pepper was successful.  And if the request was successful, we need to make sure we create the matching record in our database for bookkeeping purposes.    And likewise, we need to return a 20x status code since the operation overall was a success, so rethrowing something here just means we'd have to catch it again at the controller.  Telling an admin "your kit request was successful but DSM included some unexpected fields in its response" isn't particularly useful--I think logging as error so it gets picked up in alerting/triage is probably best because chances are the error is going to be very inside-baseball.

Reply:
we don't yet have alerting -- just the weekly log review.  We'll likely want to up that to daily and add alerting once OurHealth traffic ramps up.